### ISSUE TITLE
Builds with MinGW on Windows fails due to Hunter dependency needed updated

### ISSUE THREAD

**Issue by Hydroque** [wheybags/freeablo#322]
I would like to contribute but the hunter cmake build functionality is failing on several levels. For example, when building SDL under Windows with MinGW the wrong defines are set up and I fail the build. I am not aware of anyway to set up SDL myself as well as the other dependencies. I tried what I could to patch the stuff but nothing worked.

If you are willing to help me out I will contribute

**Comment by Hydroque** [comment]
So I decided to attempt to build SDL myself. I noticed that the wrong package gets downloaded per your compiler. There are binaries already configured for different build varieties such as MinGW, Linux, etc. So thats whats going on with that.

**Comment by Predelnik** [comment]
So it looks like current version of SDL available through Hunter is indeed fails to compile with MinGW (https://github.com/hunter-packages/SDL2/issues/1) and SDL2 current master version compiles fine. The proper fix is to update Hunter fork to be closer to its upstream. Sadly it happens to be not trivial but doable task and I've started some movement in that direction. Hopefully it will be resolved in a couple of days.

As for building it yourself I'm not Hunter expert but I'm not sure that it's supported currently. Maybe you can do it with some CMakeLists.txt fixes but I would recommend against that.

As a sidenote I would like to add that Visual Studio 2015/2017 versions should both compile Freeablo fine currently. If it's possible option for you you may also go this path.

**Comment by wheybags** [comment]
Yup, on Windows use vs. The hunter SDL2 package doesn't support mingw. The
main reason is the terrible upstream cmake files. It can and should be
fixed, but noones done so yet.
There's also still some stuff linked into SDL that is dependent on the
environment it's built in on Linux, so that needs to be fixed too

**Comment by Hydroque** [comment]
I got it to build under msys2 and I encountered a further problem with SDL_Image, which was failure to extract - so updated tools needed to be issued. However, I decided to drop it. I have an emulator for linux.

**Comment by wheybags** [comment]
Again, vs2015 will work just fine, including the free community edition. The hunter packages for all the bits of SDL2 explicitly don't work with mingw yet. Anyway, however you do get started, I'm looking forward to it, always nice to have some more help :)
I've been meaning to make a bunch of GitHub issues for new people to work on, I'll try to get that done tomorrow so you can have an idea of where to start (unless you've already got one ofc :p)

**Comment by Hydroque** [comment]
vs2015 works, yes, but issue stays open until we can confirm it works and probably is necessary to let users know that it doesn't work with mingw... I haven't tried clang. I find vs cancerous. I've been programming everything in notepad++ for 2-4 years now (that and eclipse). I did manage to complete a valid install with mingw of SDL2, but hunter fails to pick it up after I `make install` so thats something else.

Before I contribute I am going to get familiar with the repository. I am also a code nit picker so you can see beautifications of code incomming as pull requests and redundancy checks.

**Comment by wheybags** [comment]
>  I did manage to complete a valid install with mingw of SDL2, but hunter fails to pick it up

You can configure a project to work both with and without hunter, but that hasn't been done, and won't be done. We just use hunter.

Also, I've created a bunch of issues now tagged with "good first issue", and a few of them are code cleanupy type things anyway, so you might want to have a look at them.

Since the original topic of this issue is an upstream problem, with an upstream issue (https://github.com/hunter-packages/SDL2/issues/1) I'll close this one

**Comment by Hydroque** [comment]
Which I am just going to download vs build tools... even though I don't want to. I will compile it with VS and have something I can actually run. My installation of debian jessie didn't go too well and I can't get video or sound anymore, which is a shame. Oh well that is just my OS building OS.

**Comment by Predelnik** [comment]
Build with mingw should mainly work after #333 is merged.
